ui/pages/confirmations/components/confirm/info/hooks/use-token-values.ts

```diff
@@ -56,9 +59,25 @@ export const useTokenValues = (transactionMeta: TransactionMeta) => {
   const fiatDisplayValue =
     fiatValue && fiatFormatter(fiatValue, { shorten: true });
 
+  const locale = useSelector(getIntlLocale);
+  const displayTransferValue = formatAmount(
+    locale,
+    new BigNumber(decodedTransferValue),
+  );
+
   return {
-    decodedTransferValue,
+    decodedTransferValue: toNonScientificString(decodedTransferValue),
+    displayTransferValue,
     fiatDisplayValue,
     pending,
   };
 };
+
+export function toNonScientificString(num: number): string {
+  if (num >= 10e-18) {
+    return num.toFixed(18).replace(/\.?0+$/u, '');
+  }
+
+  // keep in scientific notation
+  return num.toString();
+}
```

Comment:
I extracted the logic into a contained function but since this function is only used on this hook I think it makes sense for it to live on the same file for now.